log_collector.py

```diff
@@ -37,62 +34,53 @@
 
 def make_dir(directory):
     if not os.path.exists(directory):
+        # noinspection PyBroadException
         try:
             os.mkdir(directory)
         except:
             logger.exception("Could not create directory %s - exiting", directory)
             sys.exit()
 
 
-def run(configured_namespace, configured_output_path):
-    global output_dir, namespace
-
-    namespace = ""
-    if configured_namespace:
-        namespace = configured_namespace
-    else:
+def run(namespace, output_dir):
+    if not namespace:
         namespace = get_namespace_from_config()
 
-    global timestr
-    global dir_name
-
-    if configured_output_path:
-        output_dir = os.path.join(configured_output_path, dir_name)
-    else:
-        output_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), dir_name)
+    if not output_dir:
+        output_dir_name = "redis_enterprise_k8s_debug_info_{}".format(TIME_FORMAT)
+        output_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), output_dir_name)
 
     make_dir(output_dir)
 
-    get_redis_enterprise_debug_info()
-    collect_cluster_info()
-    collect_resources_list()
-    collect_events()
-    collect_api_resources()
-    collect_pods_logs()
-    archive_files()
+    get_redis_enterprise_debug_info(namespace, output_dir)
+    collect_cluster_info(output_dir)
+    collect_resources_list(output_dir)
+    collect_events(namespace, output_dir)
+    collect_api_resources(namespace, output_dir)
+    collect_pods_logs(namespace, output_dir)
+    archive_files(output_dir, output_dir_name)
     logger.info("Finished Redis Enterprise log collector")
 
 
-def get_redis_enterprise_debug_info():
+def get_redis_enterprise_debug_info(namespace, output_dir):
     """
         Connects to an RS cluster node, creates and copies debug info package from the pod
     """
-    pod_names = get_pod_names(selector='redis.io/role=node')
+    pod_names = get_pod_names(namespace, selector='redis.io/role=node')
     if not pod_names:
         logger.warning("Cannot find redis enterprise pod")
         return
 
     pod_name = pod_names[0]
 
-    cmd = "kubectl {} exec {} /opt/redislabs/bin/rladmin cluster debug_info path /tmp".format(
-        get_namespace_argument(), pod_name)
+    cmd = "kubectl -n {} exec {} /opt/redislabs/bin/rladmin cluster debug_info path /tmp".format(namespace, pod_name)
```

Comment:
Since we have two containers, this will alway produce a warning. We might consider checking if we are in old or new version, and adding the container name if we are in the new version

Reply:
yeah that makes sense. can you ticket this?